example/32_batched_gemm_scale_softmax_gemm/grouped_multihead_attention_backward_v1.cpp

```diff
@@ -0,0 +1,1051 @@
+    std::vector<void*> p_z_nullptr; // for time test
+    std::vector<const void*> p_v;
+    std::vector<const void*> p_y;
+    std::vector<const void*> p_lse;
+    std::vector<void*> p_qgrad;
+    std::vector<void*> p_kgrad;
+    std::vector<void*> p_vgrad;
+    std::vector<const void*> p_ygrad;
+
+    std::vector<Tensor<InputDataType>> q_g_m_ks;
+    std::vector<Tensor<InputDataType>> k_g_n_ks;
+    std::vector<Tensor<ZDataType>> z_g_m_ns;
+    std::vector<Tensor<InputDataType>> v_g_n_os;
+    std::vector<Tensor<AccDataType>> s_g_m_ns;
+    std::vector<Tensor<InputDataType>> p_g_m_ns;
+    std::vector<Tensor<InputDataType>> y_g_m_os;
+    std::vector<Tensor<LSEDataType>> lse_g_ms;
+    std::vector<Tensor<InputDataType>> p_drop_g_m_ns;
+
+    std::vector<Tensor<InputDataType>> q_tensors;
+    std::vector<Tensor<InputDataType>> k_tensors;
+    std::vector<Tensor<InputDataType>> v_tensors;
+    std::vector<Tensor<InputDataType>> y_tensors;
+    std::vector<Tensor<ZDataType>> z_tensors;
+    std::vector<Tensor<LSEDataType>> lse_tensors;
+    std::vector<Tensor<OutputDataType>> qgrad_tensors;
+    std::vector<Tensor<OutputDataType>> kgrad_tensors;
+    std::vector<Tensor<OutputDataType>> vgrad_tensors;
+    std::vector<Tensor<InputDataType>> ygrad_tensors;
+
+    std::vector<DeviceMemPtr> q_tensors_device;
+    std::vector<DeviceMemPtr> k_tensors_device;
+    std::vector<DeviceMemPtr> z_tensors_device;
+    std::vector<DeviceMemPtr> v_tensors_device;
+    std::vector<DeviceMemPtr> y_tensors_device;
+    std::vector<DeviceMemPtr> lse_tensors_device;
+    std::vector<DeviceMemPtr> qgrad_tensors_device;
+    std::vector<DeviceMemPtr> ygrad_tensors_device;
+    std::vector<DeviceMemPtr> kgrad_tensors_device;
+    std::vector<DeviceMemPtr> vgrad_tensors_device;
+    std::size_t group_count = 10;
+    std::size_t flop = 0, num_byte = 0;
+    for(std::size_t i = 0; i < group_count; i++)
+    {
+        int M  = 128 * (rand() % 8) + (rand() % 128);
+        int N  = 128 * (rand() % 8) + (rand() % 128);
+        int K  = DIM;
+        int O  = DIM;
+        int G0 = rand() % 4 + 1;
+        int G1 = rand() % 4 + 1;
+        std::vector<ck::index_t> q_gs_ms_ks_lengths{G0, G1, M, K};
+        std::vector<ck::index_t> q_gs_ms_ks_strides =
+            input_permute
+                ? std::vector<ck::index_t>{M * G1 * K, K, G1 * K, 1} // Q layout [G0, M, G1, K]
+                : std::vector<ck::index_t>{G1 * M * K, M * K, K, 1}; // Q layout [G0, G1, M, K]
+
+        std::vector<ck::index_t> k_gs_ns_ks_lengths{G0, G1, N, K};
+        std::vector<ck::index_t> k_gs_ns_ks_strides =
+            input_permute
+                ? std::vector<ck::index_t>{N * G1 * K, K, G1 * K, 1} // K layout [G0, N, G1, K]
+                : std::vector<ck::index_t>{G1 * N * K, N * K, K, 1}; // K layout [G0, G1, N, K]
+
+        std::vector<ck::index_t> v_gs_os_ns_lengths{G0, G1, O, N};
+        std::vector<ck::index_t> v_gs_os_ns_strides =
```

Comment:
Suggest to unify the style of the lengths/strides containers (`v_gs_os_ns_lengths` => `v_gs_ns_os_lengths`) in all the device operators and examples for `query`, `key`, `value` and `output` tensor description